lm_human_preference_details/train_reward_jax.py

```diff
@@ -402,11 +401,21 @@ def normalize(
     generation_config,
 ):
     # number of minibatches for computing the normalization statistics
-    n_batches = ceil_div(args.local_normalize_samples, args.rollout_batch_size)
+    n_batches = ceil_div(args.normalize_samples, args.rollout_batch_size)
```

Comment:
This should still be `local_normalize_samples`, right?

Reply:
Thanks! I think `ceil_div(args.normalize_samples, args.rollout_batch_size)` can be correct? But let me know if I've overlooked anything:

In the jax version the dataloder `normalization_dataloader` returns `args.rollout_batch_size` samples per iteration. The `args.rollout_batch_size` is the number of samples for all devices. For this reason, if we want the model to see `args.normalize_samples` samples in total from all devices, we need
`ceil_div(args.normalize_samples, args.rollout_batch_size)` batches.

The situation is different in `train_reward_accelerate` though. There, we do `normalization_dataloader = accelerator.prepare(normalization_dataloader)`. In that case, I think the `args.rollout_batch_size` argument of dataloader refers to the number of samples per device. So we set `n_batches = ceil_div(args.local_normalize_samples, args.rollout_batch_size)` in that case.

Both versions should be equivalent now, although the per-device batch size is different. If we want to collect normalization statistics with the same batch size per device, perhaps we can define a `local_rollout_batch_size`? I think that with `local_rollout_batch_size`, we don't need `args.local_normalize_samples` anymore.

Reply:
Yep, just added `args.local_rollout_batch_size`. Both versions should be consistent as before. The difference at the implementation level is the `batch_size` we pass to the dataloaders:

- in pytorch:
```python
normalization_dataloader = DataLoader(normalization_dataset, batch_size=args.local_rollout_batch_size)
reward_model, optimizer, normalization_dataloader = accelerator.prepare(reward_model, optimizer, normalization_dataloader)
```

- in jax:
```python
normalization_dataloader = DataLoader(normalization_dataset, batch_size=args.rollout_batch_size)
```